### PR TITLE
Array sorting cheat sheet: Fix typos and add more sorting techniques

### PR DESCRIPTION
**Pull request by kumarankit0411**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [ ] New Instant Answer
    - [ ] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [x] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {Issue number or one-line description}`**
    - [x] Enhancement: **`{array_sorting_cheat_sheet}: Removed an extra bracket from worst case complexity of heap sort. Also added more sorting techniques(TimSort, TreeSort, CountingSort, CubeSort). Also changed the template type from "reference" to "code" which overall improves the layout and view. `**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

What changes does this PR introduce? 

###### Related Issues or Discussions 

What issues or discussions (if any) does this PR address? Please provide the issue # or link.

###### People to notify

@mention any relevant people/organizations

---

Instant Answer Page: https://duck.co/ia/view/array_sorting_cheat_sheet
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mintsoft

### CODE REVIEW

**Comment by daxtheduck** [comment]
@kumarankit0411 Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/array-sorting.json [hunk 1]
```diff
@@ -7,110 +7,207 @@
...
+        "array sorting algorithms", "array sorting algorithm", "sorting complexity"
+
     ],
-    "template_type": "reference",
+    "template_type": "code",
     "section_order": [
         "Quicksort",
         "Mergesort",
         "Heapsort",
+        "TimSort",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/array-sorting.json]: I think we should be consistent on the casing; i.e. "Timsort" for 1 word or "Tim Sort" for 2 words. IIRC Timsort is 1 word

File: share/goodie/cheat_sheets/json/array-sorting.json [hunk 2]
```diff
@@ -7,110 +7,207 @@
...
         "Bubble Sort",
         "Insertion Sort",
         "Selection Sort",
+        "TreeSort",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/array-sorting.json]: Same here I think this should be "Tree Sort"

File: share/goodie/cheat_sheets/json/array-sorting.json [hunk 3]
```diff
@@ -7,110 +7,207 @@
...
         "Shell Sort",
         "Bucket Sort",
-        "Radix Sort"
+        "Radix Sort",
+        "CountingSort",
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/array-sorting.json]: "Counting Sort" I think

File: share/goodie/cheat_sheets/json/array-sorting.json [hunk 4]
```diff
@@ -7,110 +7,207 @@
...
+        "CubeSort"
```
Comment (mintsoft) [inline share/goodie/cheat_sheets/json/array-sorting.json]: "Cubesort" I think

**Comment by mintsoft** [comment]
@kumarankit0411 Apologies for the delay here, I've put some minor feedback against the diff

**Comment by kumarankit0411** [comment]
@mintsoft Thanks for pointing out casing issue. I have corrected it.

**Comment by mintsoft** [comment]
Cool thanks for the quick fix @kumarankit0411 This LGTM

**Pull request merged**